src/networkpolicy/networkPolicy.go

```diff
@@ -2133,6 +2137,52 @@ func PopulateNetworkPoliciesFromNetworkLogs(networkLogs []types.KnoxNetworkLog)
 	return discoveredNetworkPolicies
 }
 
+func writeNetworkPoliciesYamlToDB(policies []types.KnoxNetworkPolicy) {
+	clusternames := []string{}
+
+	for _, pol := range policies {
+		clusternames = append(clusternames, pol.Metadata["cluster_name"])
+	}
+
+	// convert knoxPolicy to CiliumPolicy
+	ciliumPolicies := plugin.ConvertKnoxPoliciesToCiliumPolicies(policies)
+
+	res := []types.Policy{}
+
+	for index, ciliumPolicy := range ciliumPolicies {
+		var label string
+
+		jsonBytes, err := json.Marshal(ciliumPolicy)
+		if err != nil {
+			log.Error().Msg(err.Error())
+			continue
+		}
+		yamlBytes, err := yaml.JSONToYAML(jsonBytes)
+		if err != nil {
+			log.Error().Msg(err.Error())
+			continue
+		}
+		policyYaml := string(yamlBytes)
+
+		for k, v := range ciliumPolicy.Spec.EndpointSelector.MatchLabels {
```

Comment:
VM/host policies should take into account here, For host policies, the labels will be available in `ciliumPolicy.Spec.NodeSelector.MatchLabels`. You should check which `MatchLabels` is empty.